### ISSUE TITLE
Escape hatch filters don't quite work

### ISSUE THREAD

**Issue by jphenow** [hashicorp/consul#11382]
#### Overview of the Issue

Looking to plugin a gRPC ext_authz filter for some specific use-cases. In trying to configure the gRPC service I've come across some strange behavior with the actual configuration serialization.

While doing this investigation I came across a test that specifically looks at making sure consul can unmarshal "escape hatch" types of configuration that people might be using. It appears an HTTP ext_authz filter works but a gRPC service explicitly does not. I'm able to reproduce the error I see ADS delta stream by adding to this tests (see below).

#### Reproduction Steps

```diff
diff --git a/agent/xds/z_xds_packages_test.go b/agent/xds/z_xds_packages_test.go
index b102dcb3a..53ca3053e 100644
--- a/agent/xds/z_xds_packages_test.go
+++ b/agent/xds/z_xds_packages_test.go
@@ -31,6 +31,21 @@ func TestUnusedExtensions(t *testing.T) {
 			  }
 			} `,
 		},
+		{
+			"type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthz",
+			` {
+				"@type": "type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthz",
+				"transport_api_version": "V3",
+				"grpc_service": {
+					"envoy_grpc": {
+						"cluster_name": "opa"
+					},
+					"timeout": "0.5s"
+				},
+				"failure_mode_allow": false,
+				"include_peer_certificate": true
+			} `,
+		},
 		{
 			"type.googleapis.com/envoy.config.trace.v3.ZipkinConfig",
 			` {
```

With this dropped in the test now fails. Envoy docs [here](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/ext_authz_filter#config-http-filters-ext-authz) and [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto#envoy-v3-api-msg-extensions-filters-http-ext-authz-v3-extauthz) seem to indicate this should work and I have had a form of this working without consul in the mix previously. 

Test failure:

```
Running tool: /usr/local/bin/go test -timeout 30s -run ^TestUnusedExtensions$ github.com/hashicorp/consul/agent/xds

--- FAIL: TestUnusedExtensions (0.00s)
    --- FAIL: TestUnusedExtensions/type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthz#01 (0.00s)
        /Users/jon.phenow/Code/consul/agent/xds/z_xds_packages_test.go:64: 
            	Error Trace:	z_xds_packages_test.go:64
            	Error:      	Received unexpected error:
            	            	can't unmarshal Any nested proto *envoy_extensions_filters_http_ext_authz_v3.ExtAuthz: unknown field "transport_api_version" in envoy_extensions_filters_http_ext_authz_v3.ExtAuthz
            	Test:       	TestUnusedExtensions/type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthz#01
FAIL
FAIL	github.com/hashicorp/consul/agent/xds	0.483s
FAIL
```

**Big note**: the core of the issue above is that `transport_api_version` can't be passed, though the [type](https://github.com/envoyproxy/go-control-plane/blob/aa41abea9ba8125b8be10a2851a645bbebd23825/envoy/extensions/filters/http/ext_authz/v3/ext_authz.pb.go#L47) allows it. Normally, I'd say "default is good in this case" but for [version compatibility](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/config_source.proto.html#enum-config-core-v3-apiversion) it defaults to "AUTO" which assumes v2...

"What do I care?" well, this results in errors and failure to update on the proxy itself:

```
. The v2 xDS major version is deprecated and disabled by default. Support for v2 will be removed from Envoy at the start of Q1 2021. You may make use of v2 in Q4 2020 by following the advice in https://www.envoyproxy.io/docs/envoy/latest/faq/api/transition.
```

tl;dr `transport_api_proxy` can't be set which renders it nearly unusable. 

---

Happy to provide more detail but system and version info seems irrelevant since this is all reproducible from a build of main.

**Comment by blake** [comment]
@jphenow Consul 1.12.0 now uses go-control-plane version 0.9.9.

Are you able to test your change with Consul 1.12 and let us know whether you're still experiencing this issue with that version?

**Comment by jphenow** [comment]
Oh boy; I am no longer working with the company where I ran into this, but am separately excited this was addressed. I will see if I can spend a bit of time constructing this scenario again this week so we can close this confidently

**Comment by david-yu** [comment]
Hi there. We now how have an ext-authz extension for Consul to configure Envoy's ext-authz filter starting with 1.16.0-rc1. Please try it out and give us feedback via a new issue. The docs also provide a reference and examples of usage: https://developer.hashicorp.com/consul/docs/v1.16.x/connect/proxies/envoy-extensions/configuration/ext-authz. I'll close this for now. Thanks for your feedback.
